### ISSUE TITLE
Python 3.4 supported SSL version

### ISSUE THREAD

**Issue by 0ac4efd1-fad3-472e-81f8-af9d1272de74** [python/cpython#76019]
BPO | [31838](https://bugs.python.org/issue31838)
--- | :---
Nosy | @tiran, @sabian2008

<sup>*Note: these values reflect the state of the issue at the time it was migrated and might not reflect the current state.*</sup>

<details><summary>Show more details</summary><p>

GitHub fields:
```python
assignee = 'https://github.com/tiran'
closed_at = <Date 2017-10-22.13:44:40.957>
created_at = <Date 2017-10-21.23:25:15.453>
labels = ['expert-SSL']
title = 'Python 3.4 supported SSL version'
updated_at = <Date 2017-10-22.13:44:40.957>
user = 'https://github.com/sabian2008'
```

bugs.python.org fields:
```python
activity = <Date 2017-10-22.13:44:40.957>
actor = 'sabian2008'
assignee = 'christian.heimes'
closed = True
closed_date = <Date 2017-10-22.13:44:40.957>
closer = 'sabian2008'
components = ['SSL']
creation = <Date 2017-10-21.23:25:15.453>
creator = 'sabian2008'
dependencies = []
files = []
hgrepos = []
issue_num = 31838
keywords = []
message_count = 3.0
messages = ['304725', '304738', '304750']
nosy_count = 2.0
nosy_names = ['christian.heimes', 'sabian2008']
pr_nums = []
priority = 'normal'
resolution = None
stage = 'resolved'
status = 'closed'
superseder = None
type = None
url = 'https://bugs.python.org/issue31838'
versions = ['Python 3.4']
```

</p></details>

**Comment by 0ac4efd1-fad3-472e-81f8-af9d1272de74** [comment]
Hello and sorry to bother. This is my first message to the list.

I'm trying to build python 3.4.7 downloaded from python.org (released in August this year) and while following the exact same steps detailed in  https://bugs.python.org/issue29027, I get getting the exact same error he does.

Doesn't Python 3.4 support OpenSSL v 1.1.0? Is this is the case, I guess it will never do, right?

More importantly, is there a place to get build dependencies (with appropriate versions) for each CPython version? I scrawled the source and the GitHub repositories without luck.

I guess this has been asked before, but after several hours, I couldn't find anything.

Cheers,
Mauro

**Comment by tiran** [comment]
You are correct, Python 3.4 does not support OpenSSL 1.1 and will never support it.

The last bug fix release of Python 3.4 was 3.4.4 in 2015-12-20. 3.4.5 and newer are security fix releases that only contain critical security updates. OpenSSL 1.1.0 was released 2016-08-25, about 9 months after Python 3.4 reached security-only mode.

We have a repository for Windows source dependencies: https://github.com/python/cpython-source-deps. It's not meant for general consumption.

**Comment by 0ac4efd1-fad3-472e-81f8-af9d1272de74** [comment]
Hi Christian

Great! Thanks for the information.

I apologize if this was trivial, just wanted to make sure.

Cheers,
